### PR TITLE
Broad organizational and bug-fix changes

### PR DESCRIPTION
**Pull request by mnadareski**
This PR encompasses multiple pieces that bring the code to a much more usable and stable state

- The solution file includes folders for organization
- Most classes and methods now have XML comments on them
- Methods have been organized into regions, where appopriate
- Classes all live in their own files
- Two new settings added: 1) Play data tracks as audio, and 2) Generate TOC if one does not exist
- General stability and usability improvements to the Player class

### CODE REVIEW

File: RedBookPlayer/MainWindow.xaml.cs [hunk 1]
```diff
@@ -19,34 +19,37 @@ public MainWindow()
...
-            if((theme ?? "") == "")
-            {
+            // If no theme path is provided, we can ignore
+            if(string.IsNullOrWhiteSpace(theme))
                 return;
-            }
 
-            if(theme == "default")
+            // If the theme name is "default", we assume the internal theme is used
+            if(theme.Equals("default", StringComparison.OrdinalIgnoreCase))
```
Comment (claunia) [inline RedBookPlayer/MainWindow.xaml.cs]: Use current culture, we can have a theme called `molón` :D

File: RedBookPlayer/Player.cs [hunk 2]
```diff
@@ -1,270 +1,372 @@
...
+                        if(CurrentSector >= trackData.TrackStartSector)
+                        {
+                            CurrentTrack = (int)trackData.TrackSequence - 1;
+                            break;
+                        }
                     }
                 }
 
-                foreach((ushort key, int i) in Image.Tracks[CurrentTrack].Indexes.Reverse())
+                foreach(KeyValuePair<ushort, int> item in track.Indexes.Reverse())
```
Comment (claunia) [inline RedBookPlayer/Player.cs]: Leave it decomposed.

File: RedBookPlayer/Player.cs [hunk 3]
```diff
@@ -1,270 +1,372 @@
...
+
+        /// <summary>
+        /// Represents the copy allowed flag
+        /// </summary>
+        public bool CopyAllowed { get; private set; } = false;
+
+        /// <summary>
+        /// Represents the track type
+        /// </summary>
+        public TrackTypeValue? TrackType { get; private set; }
```
Comment (claunia) [inline RedBookPlayer/Player.cs]: Is there a reason to not use `Aaru.CommonTypes.TrackType` ?

File: RedBookPlayer/Player.cs [hunk 4]
```diff
@@ -453,46 +563,298 @@ public void Rewind()
...
+                        Frame = leadoutAmsf.frame,
+                        PHOUR = 2,
+                        PMIN = leadoutPmsf.minute,
+                        PSEC = leadoutPmsf.second,
+                        PFRAME = leadoutPmsf.frame
+                    });
+
+                    // This seems to be constant? It should not exist on CD-ROM but CloneCD creates them anyway
+                    // Format seems like ATIP, but ATIP should not be as 0xC0 in TOC...
+                    trackDescriptors.Add(new TrackDataDescriptor
```
Comment (claunia) [inline RedBookPlayer/Player.cs]: You don't need C0 point.

**Review by claunia — CHANGES_REQUESTED** [review]

File: RedBookPlayer/Player.cs (hunk 3 above)
Comment (mnadareski) [reply RedBookPlayer/Player.cs]: I was just extracting the current enum to a new place. I can do that in a future PR, unless you feel strongly about me changing this now.

File: RedBookPlayer/Player.cs (hunk 4 above)
Comment (mnadareski) [reply RedBookPlayer/Player.cs]: This entire block was copied whole from CloneCd, so I'll admit I don't fully understand how it all works. I'll comment out that block.

**Review by claunia — APPROVED** [review]

**Pull request merged**